migrations/20220818164120_create_anchor_and_request.js

```diff
@@ -0,0 +1,102 @@
+const REQUEST_TABLE_NAME = 'request'
+const TEMP_TABLE_NAME = 'temp'
+const ANCHOR_TABLE_NAME = 'anchor'
+
+const sortByStringKeys = (obj) =>
+  Object.fromEntries(Object.entries(obj).sort(([a], [b]) => a.localeCompare(b)))
+
+const checkObjEquality = (expected, received) =>
+  JSON.stringify(sortByStringKeys(expected)) === JSON.stringify(sortByStringKeys(received))
+
+const verifyTableStructure = async (knex, tableWithValidStructure, tableToCheck) => {
+  const [expectedStructure, existingStructure] = await Promise.all([
+    knex(tableWithValidStructure).columnInfo(),
+    knex(tableToCheck).columnInfo(),
+  ])
+  const structuresAreEqual = checkObjEquality(expectedStructure, existingStructure)
+  if (!structuresAreEqual) {
+    throw Error(`Existing ${tableToCheck} table does not have the expected structure`)
+  }
+}
+
+const createRequestTable = async (knex, tableName) => {
+  await knex.schema.createTable(tableName, (table) => {
+    table.uuid('id').notNullable().unique().primary().defaultTo(knex.raw('uuid_generate_v4()'))
+    table.integer('status').notNullable()
+    table.boolean('pinned').notNullable().defaultTo(false)
+    table.timestamp('created_at', { useTz: false }).notNullable().defaultTo(knex.raw('now()'))
+    table.timestamp('updated_at', { useTz: false }).notNullable().defaultTo(knex.raw('now()'))
+  })
+
+  // knex defaults to varchar(255).
```

Comment:
I think this is what's happening in the ComposeDB database tables currently.  @willex can you follow up on this?

Reply:
Yes, very similar in terms of implementation as well. Added the note about the temp table creation vs. shipping the table structure as an object.

**Minor nitpicks:**
- would be better to store with time zone so we enforce UTC
- instead of raw you can use the built in `now()` function

**Example:**
`table.dateTime('updated_at').notNullable().defaultTo(dbConnection.fn.now())`

results in:
`updated_at        | timestamp with time zone |           | not null | CURRENT_TIMESTAMP`

Reply:
Yes i agree with these minor nitpicks. I did these weird things to keep as close to the current table structures as possible as to not interfere with anything. I ideally would like to use `timestamps` which includes a trigger. The interferences I can think of that may or may not appear is logging, metrics, related things. If these don't seem like a big deal i'll change things to timestamps and implement the shipping the table structure as an object